handlers/geolocation/geolocation.go

```diff
@@ -26,29 +27,51 @@ type GeolocationHandlersCollection struct {
 
 // redirect an incoming user to a node for playback or 404 handling
 func (c *GeolocationHandlersCollection) RedirectHandler() httprouter.Handle {
-	return func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-		nodeHost := c.Config.NodeHost
-		redirectPrefixes := c.Config.RedirectPrefixes
+	return func(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+		host := r.Host
+		prefix, playbackID, pathTmpl, isValid := parsePlaybackID(r.URL.Path)
+
+		if c.Config.CdnRedirectPrefix != "" {
+
+			if !isValid {
```

Comment:
I'm not sure what's the intention of [these tests](https://github.com/livepeer/catalyst-api/blob/main/handlers/geolocation/gelocation_test.go#L280-L300) they test redirects without playbackID in the URL path.
I don't understand if we want to make sure that redirect works for whatever path or should we always expect playbackID.

Reply:
@pwilczynskiclearcode The `nodeHost` stuff is the 404 handler! The thinking is this: `catalyst-node-1` crashes, but nginx is still online so the request gets handled by a different catalyst in the same cluster. So then this happens:

```
> GET https://catalyst-node-1/hls/video+foo/index.m3u8
(catalyst-node-2 thinks: uhhh, that's not me, I only handle requests for nodeHost=playback.livepeer.studio, let's send them there)
< 302 Location: https://playback.livepeer.studio/hls/video+foo/index.m3u8

> GET https://playback.livepeer.studio/hls/video+foo/index.m3u8
(catalyst-node-2 thinks: that's better! Let's send them to the regular geolocation loop... looks like catalyst-node-5 isn't handling much traffic right now.)
< 302 Location: https://catalyst-node-5/hls/video+foo/index.m3u8

> GET https://catalyst-node-5/hls/video+foo/index.m3u8
< 200 OK
```

I coded that process to work for any arbitrary URL just in case. If it needs to work with just a playback id, that's okay too.